pkgs/development/libraries/xeus-zmq/default.nix

```diff
@@ -0,0 +1,38 @@
+{ clangStdenv
+, cmake
+, fetchFromGitHub
+, lib
+
+# Dependency libraries
+, cppzmq
+, libuuid
+, nlohmann_json
+, openssl
+, xeus
+, xtl
+, zeromq
+}:
+
+clangStdenv.mkDerivation rec {
+  pname = "xeus-zmq";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "jupyter-xeus";
+    repo = "xeus-zmq";
+    rev = "${version}";
+    sha256 = "0m9g0fqm001d0jk91yk139d42azcinzhkqhaw7qhx3dh18zcsvcg";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ cppzmq openssl libuuid xeus xtl zeromq ];
+  propagatedBuildInputs = [ nlohmann_json ];
```

Comment:
```suggestion
  nativeBuildInputs = [ cmake ];
  
  buildInputs = [ 
    cppzmq 
    libuuid
    openssl 
    xeus
    xtl
    zeromq 
  ];
  
  propagatedBuildInputs = [ nlohmann_json ];
```

Reply:
Same question -- I can readily find examples of `buildInputs` not being full of newlines in Nixpkgs. Is this necessary? (Is there a style guide or something out there?)

Reply:
We don't have a style guide yet - there is a RFC in the queue: https://github.com/NixOS/rfcs/pull/101

The rationale here is to minimize the line count of  future patches.
If you separate the elements of a list per line, then future modifications will be clearly highlighted:
- A new dependency will be a green line with a plus sign
- A removed dependency will be a red line with a minus sign

On the other hand, putting everything on the same line removes such clues, generating a red and a green line.

Reply:
Also, Nixpkgs is an old codebase. Many things need to be modernized, few people are interested on such a huge task.
Indeed I have such old and ugly code that I am reviewing now.